core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/withEditor.ts

```diff
@@ -0,0 +1,306 @@
+                    patchState(store, { state: EDITOR_STATE.SCROLL_DRAG, bounds: [] });
+                },
+                setEditorState(state: EDITOR_STATE) {
+                    patchState(store, { state: state });
+                },
+                setEditorDragItem(dragItem: EmaDragItem) {
+                    patchState(store, { dragItem, state: EDITOR_STATE.DRAGGING });
+                },
+                setEditorContentletArea(contentletArea: ContentletArea) {
+                    const currentContentletArea = store.contentletArea();
+
+                    if (
+                        currentContentletArea?.x === contentletArea.x &&
+                        currentContentletArea?.y === contentletArea.y
+                    ) {
+                        // Prevent updating the state if the contentlet area is the same
+                        // This is because in inline editing, when we select to not copy the content and edit global
+                        // The contentlet area is updated on focus with the same values and IDLE
+                        // Losing the INLINE_EDITING state and making the user to open the dialog for checking whether to copy the content or not
+                        // Which is an awful UX
+
+                        return;
+                    }
+
+                    patchState(store, {
+                        contentletArea: contentletArea,
+                        state: EDITOR_STATE.IDLE
+                    });
+                },
+                setEditorBounds(bounds: Container[]) {
+                    patchState(store, { bounds });
+                },
+                resetEditorProperties() {
+                    patchState(store, {
+                        dragItem: null,
+                        contentletArea: null,
+                        bounds: [],
+                        state: EDITOR_STATE.IDLE
+                    });
+                },
+                getPageSavePayload(positionPayload: PositionPayload): ActionPayload {
+                    const { containers, languageId, id, personaTag } = store.$pageData();
+
+                    const { contentletsId } = containers.find((container) =>
+                        areContainersEquals(container, positionPayload.container)
+                    ) ?? { contentletsId: [] };
+
+                    const container = positionPayload.container
+                        ? {
+                              ...positionPayload.container,
+                              contentletsId
+                          }
+                        : null;
+
+                    return {
+                        ...positionPayload,
+                        language_id: languageId.toString(),
+                        pageId: id,
+                        pageContainers: containers,
+                        personaTag,
+                        container
+                    };
+                },
+                getCurrentTreeNode(
```

Comment:
This can be a `computed`

Reply:
it cannot be because we take a payload from an event and not from the store